autogluon/task/tabular_prediction/predictor.py

```diff
@@ -688,6 +688,75 @@ def load_data_internal(self, dataset='train', return_X=True, return_y=True):
         return X, y
 
+    def reduce_memory_size(self, remove_data=True, remove_fit_stack=True, requires_save=True, reduce_children=False):
+        """
+        Reduces the memory and disk size of `predictor`.
+        Note that most methods of reducing `predictor` size involve removing advanced functionality.
+
+        Parameters
+        ----------
+        remove_data : bool, default = True
+            Whether to remove cached files of the original training and validation data.
+            Only reduces disk usage, it has no impact on memory usage.
+            This is especially useful when the original data was large.
+            This is equivalent to setting `cache_data=False` during the original `fit()`.
+                Will disable all advanced functionality that requires `cache_data=True`.
+        remove_fit_stack : bool, default = True
+            Whether to remove information required to fit new stacking models and continue fitting bagged models with new folds.
+            Only reduces disk usage, it has no impact on memory usage.
+            This includes:
+                out-of-fold (OOF) predictions
+            This is useful for multiclass problems with many classes, as OOF predictions can become very large on disk. (1 GB per model in extreme cases)
+            This disables `predictor.refit_full()` for stacker models.
+        requires_save : bool, default = True
+            Whether to remove information that requires the model to be saved again to disk.
+            Typically this only includes flag variables that don't have significant impact on memory or disk usage, but should technically be updated due to the removal of more important information.
+                An example is the `is_data_saved` boolean variable in `trainer`, which should be updated to `False` if `remove_data=True` was set.
+        reduce_children : bool, default = False
+            Whether to apply the reduction rules to bagged ensemble children models.
+            This should generally be kept as `False` since the most important memory reduction techniques are automatically applied to these models during the original `fit()` call.
+
+        """
+        self._trainer.reduce_memory_size(remove_data=remove_data, remove_fit_stack=remove_fit_stack, remove_fit=True, remove_info=False, requires_save=requires_save, reduce_children=reduce_children)
+
+    def delete_models(self, models_to_keep=None, models_to_delete=None, allow_delete_cascade=False, delete_from_disk=True, dry_run=True):
+        """
+        Deletes models from `predictor`.
+        This can be helpful to minimize memory usage and disk usage, particularly for model deployment.
+        This will remove all references to the models in `predictor`.
+            For example, removed models will not appear in `predictor.leaderboard()`.
+
+        Parameters
+        ----------
+        models_to_keep : str or list, default = None
+            Name of model or models to not delete.
```

Comment:
To prevent FAQ, may consider adding:
"To see the list of possible model-names, use: `Predictor.model_names` or `Predictor.leaderboard()`

More generally may want to consider adding this clarification in docs of every Predictor method that takes in model-names, as I otherwise foresee this being FAQ

Reply:
Added. At a certain point we will have to figure out if endlessly repeating that statement is the ideal way to communicate to users.

Reply:
> Added. At a certain point we will have to figure out if endlessly repeating that statement is the ideal way to communicate to users.

Ya not sure what's the best to address it but I think it'll be a question many people have. Maybe just having a dedicated notebook in the tutorials on various operations with one particular model will suffice.